addons/timer/models/timer_mixin.py

```diff
@@ -0,0 +1,120 @@
+            ('res_id', '=', self.id),
+            ('res_model', '=', self._name)
+        ], limit=1)
+
+    @api.model
+    def _get_user_timers(self):
+        # Get the running timer of a user
+        # Return a singleton
+        return self.env['timer.timer'].search([('user_id', '=', self.env.user.id)])
+
+    def action_timer_start(self):
+        """ Start the timer of the current record
+        First, if a timer is running, stop or pause it
+        If there isn't a timer for the current record, create one then start it
+        Otherwise, resume or start it
+        """
+        self.stop_timer_in_progress()
+        timer = self._get_record_timer()
+        if not timer:
+            timer = self.env['timer.timer'].create({
+                'timer_start' : False,
+                'timer_pause' : False,
+                'is_timer_running' : False,
+                'res_model' : self._name,
+                'res_id' : self.id,
+                'user_id' : self.env.user.id,
+            })
+            timer.action_timer_start()
+        else:
+            # Check if it is in pause then resume it or start it
+            if timer.timer_pause:
+                timer.action_timer_resume()
+            else:
+                timer.action_timer_start()
+        self._compute_timer()
+        
+
+    def action_timer_stop(self):
+        """ Stop the timer of the current record
+        Unlink the timer, it's useless to keep the stopped timer.
+        A new timer can be create if needed
+        Return the amount of minutes spent
+        """
+        timer = self._get_record_timer()
+        minutes_spent = timer.action_timer_stop()
+        self.env['timer.timer'].search([
+                    ('id', '=', timer.id)
+                ]).unlink()
+        self._compute_timer()
+        return minutes_spent
+        
+
+    def action_timer_pause(self):
+        self.ensure_one()
+        timer = self._get_record_timer()
+        timer.action_timer_pause()
+        self._compute_timer()
+
+    def action_timer_resume(self):
+        self.ensure_one()
+        self.stop_timer_in_progress()
+        timer = self._get_record_timer()
+        timer.action_timer_resume()
+        self._compute_timer()
+    
+    def interruption(self):
+        # Interruption is the action called when the timer is stoped by the start of another one
+        self.action_timer_pause()
+
+    def stop_timer_in_progress(self):
+        """
+        Cancel the timer in progress if there is one
+        Each model can interrupt the running timer in a specific way
+        By setting it in pause or stop by example
+        """
+        # The loop will be trigered only one time because only one timer
+        # can be running at the same time
+        for timer in self._get_user_timers().filtered(lambda t: t.is_timer_running):
+            model = self.env[timer.res_model].search([
+                ('id', '=', timer.res_id)
+            ])            
```

Comment:
@LucasLefevre Not everytime because we can stop a timer which has be started on another model.
We could be in hr_timesheet (account.analytic.line). But I already have a timer running in task (project.task). So, self != model in that case

Reply:
oh, right! But the search is still not needed, just browse the record ;) 
`self.env[timer.res_model].browse(timer.res_id)`